ec/ecdatasource/deploymentsdatasource/schema.go

```diff
@@ -62,65 +79,133 @@ func (d *DataSource) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnost
 				MarkdownDescription: "Unique identifier of this data source.",
 			},
 			"return_count": {
-				Type:     types.Int64Type,
-				Computed: true,
+				Type:        types.Int64Type,
+				Description: "The number of deployments actually returned.",
+				Computed:    true,
 			},
 			"deployments": deploymentsListSchema(),
-
+		},
+		Blocks: map[string]tfsdk.Block{
 			// Deployment resources
-			"elasticsearch":       resourceFiltersSchema(),
-			"kibana":              resourceFiltersSchema(),
-			"apm":                 resourceFiltersSchema(),
-			"integrations_server": resourceFiltersSchema(),
-			"enterprise_search":   resourceFiltersSchema(),
+			"elasticsearch":       resourceFiltersSchema(Elasticsearch),
```

Comment:
I think it makes sense to implement slice `names` from `resourceFiltersSchema` as a method of the type `ResourceKind`, e.g. `func (rk ResourceKind) Name() string`. It should be more safe, e.g. index cannot go beyond the range and the method can provide default "unknown" value just in case something goes wrong.

Reply:
This is implemented now.